frontend/lib/uast/post-parse-checks.cpp

```diff
@@ -1946,6 +1948,22 @@ void Visitor::checkInheritExprValid(const AstNode* node) {
   }
 }
 
+void Visitor::checkIterNames(const Function* node) {
+  if (node->kind() == Function::ITER && node->isMethod()) {
+    auto name = node->name();
+
+    if (name == USTR("init")) {
+      error(node,
+            "iterators can't be initializers, please rename this iterator");
+    } else if (chpl::parsing::isSpecialMethodName(name)) {
+      error(node,
+            "iterators can't be the %s method, please rename this iterator",
```

Comment:
Let's put the `%s` in quotes so we get e.g., "iterators can't be the 'deserialize' method, please..."?

Reply:
Oh, and should we cover `init=`?
Oh, and should we special case `deinit`, `postinit`, `init=` as was done for `init`?

I'm half a mind to do that or just use the `%s` branch for everything, up to you.

Reply:
`init=` is a good call, though I'm not sure if it's even possible to type without `proc` . . . it could very well be a syntax error.  I'll check!

I had the specialized message for initializers because that seemed like it justified a unique message, but I didn't feel as strongly about putting `deinit` and `postinit` under the same bucket.  `deinit` at least has "deinitializer" as a term, so I could do that one independently.

Reply:
Sounds good.